src/components/sign-in-form.jsx

```diff
@@ -1,36 +1,76 @@
 "use strict";
 
 import React from "react";
-import { Button } from "react-bootstrap";
-import { Form,
-         CharField as charField,
-         EmailField as emailField,
-         PasswordInput,
-         RenderForm } from "newforms";
-
-
-let SignInFormSchema = Form.extend({
-  email: emailField({required: true}),
-  password: charField({widget: PasswordInput, required: true})
-});
 
+import { ButtonInput, Input, ListGroup, ListGroupItem } from "react-bootstrap";
 
 export default React.createClass({
   displayName: "SignInForm",
 
+  propTypes: {
+    onSignIn: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps() {
+    return {
+      hasErrored: false
+    };
+  },
+
+  getInitialState() {
+    return {
+      email: "",
+      password: "",
+      hasErrored: this.props.hasErrored
+    };
+  },
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      hasErrored: nextProps.hasErrored
+    });
+  },
+
   render() {
-    return <form onSubmit={this.onSubmit}>
-      <RenderForm form={SignInFormSchema} ref="signInForm" />
-      <Button bsStyle="primary" onClick={this.onSubmit}>Loghează-te</Button>
-    </form>;
+    return <div>
+      <form onSubmit={this.onSubmit}>
+        <Input type="text"
+               label="Adresa de email"
+               onChange={this.onEmailChange} />
+        <Input type="password"
+               label="Parola"
+               onChange={this.onPasswordChange} />
+        <ButtonInput type="submit" bsStyle="primary">Loghează-te</ButtonInput>
+      </form>
+      { this.state.hasErrored ? this.renderError() : null }
```

Comment:
Cel mai usor cand faci code review pe React ca sa intelegi flow-ul faci grep dupa `this.setState`.

https://github.com/infoeducatie/infoeducatie-ui/blob/46b152d719d8f18d34a84d0438e3e99c1ac55bbb/src/components/sign-in.jsx#L60